reproducibility_project/src/engines/hoomd/project.py

```diff
@@ -104,45 +170,90 @@ def run_hoomd(job):
+            filter=hoomd.filter.All(), kT=kT, tau=1000 * dt
+        )
+    integrator.methods = [integrator_method]
     sim.operations.integrator = integrator
     sim.state.thermalize_particle_momenta(filter=hoomd.filter.All(), kT=kT)
 
-    # Shrink step follows this example
-    # https://hoomd-blue.readthedocs.io/en/latest/tutorial/
-    # 01-Introducing-Molecular-Dynamics/03-Compressing-the-System.html
-    ramp = hoomd.variant.Ramp(A=0, B=1, t_start=sim.timestep, t_ramp=int(2e4))
-    initial_box = sim.state.box
-    L = job.sp.box_L_liq
-    final_box = hoomd.Box(Lx=L, Ly=L, Lz=L)
-    box_resize_trigger = hoomd.trigger.Periodic(10)
-    box_resize = hoomd.update.BoxResize(
-        box1=initial_box,
-        box2=final_box,
-        variant=ramp,
-        trigger=box_resize_trigger,
-    )
-    sim.operations.updaters.append(box_resize)
-    sim.run(2e4 + 1)
-    assert sim.state.box == final_box
-    sim.operations.updaters.remove(box_resize)
+    if method == "npt":
+        # only run with high tauS if we are starting from scratch
+        if not restart:
+            sim.run(1e6)
+        integrator.tauS = 500 * dt
+        print(f"tauS: {sim.operations.integrator.tauS}")
+    else:
+        if not restart:
+            # Shrink step follows this example
+            # https://hoomd-blue.readthedocs.io/en/latest/tutorial/
+            # 01-Introducing-Molecular-Dynamics/03-Compressing-the-System.html
+            ramp = hoomd.variant.Ramp(
+                A=0, B=1, t_start=sim.timestep, t_ramp=int(2e4)
+            )
+            # the target volume should be the average volume from NPT
+            target_volume = job.doc.avg_volume
+            initial_box = sim.state.box
+            L = target_volume ** (1 / 3)
+            final_box = hoomd.Box(Lx=L, Ly=L, Lz=L)
+            box_resize_trigger = hoomd.trigger.Periodic(10)
+            box_resize = hoomd.update.BoxResize(
+                box1=initial_box,
+                box2=final_box,
+                variant=ramp,
+                trigger=box_resize_trigger,
+            )
+            sim.operations.updaters.append(box_resize)
+            sim.run(2e4 + 1)
+            assert sim.state.box == final_box
+            sim.operations.updaters.remove(box_resize)
 
     sim.run(1e6)
-    job.doc.finished = True
+    job.doc[f"{method}_finished"] = True
+
+
+def check_equilibration(job, method, eq_property):
+    """Check whether a simulation is equilibrated."""
+    import numpy as np
+
+    import reproducibility_project.src.analysis.equilibration as eq
+
+    data = np.genfromtxt(job.fn(f"log-{method}.txt"), names=True)
+    prop_data = data[eq_property]
+    iseq, _, _, _ = eq.is_equilibrated(prop_data)
+    if iseq:
+        uncorr, i, g, N = eq.trim_non_equilibrated(prop_data)
+        job.doc[f"avg_{eq_property}"] = np.average(uncorr)
+        job.doc[f"std_{eq_property}"] = np.std(uncorr)
+    job.doc[f"{method}_eq"] = iseq
+    return iseq
```

Comment:
I dont think this is doing what we expect.

`trim_non_equilibrated` returns the dataset `a_t` that begins from `t0`, so you still need to run `subsampleCorrelatedData` on this. At least for the avg properties? Or maybe this is just a quick check?

Reply:
oops! you are correct. I will fix this now